src/ntfsck.c

```diff
@@ -2613,6 +2567,54 @@ static int ntfsck_scan_index_entries_btree(ntfs_volume *vol)
 		vol->lost_found = lf_ni->mft_no;
 		ntfs_inode_close(lf_ni);
 	}
+}
+
+static ntfs_inode *ntfsck_check_root_inode(ntfs_volume *vol)
+{
+	ntfs_inode *ni;
+
+	ni = ntfs_inode_open(vol, FILE_root);
+	if (!ni) {
+		ntfs_log_error("Couldn't open the root directory.\n");
+		goto err_out;
+	}
+
+	ntfs_inode_attach_all_extents(ni);
+
+	if (ntfsck_check_directory(ni)) {
+		ntfs_log_error("Root directory has corrupted.\n");
+		exit(-1);
+	}
+
+	ntfsck_set_mft_record_bitmap(ni);
+	return ni;
+
+err_out:
+	if (ni)
+		ntfs_inode_close(ni);
+	return NULL;
+}
+
+static int ntfsck_scan_index_entries_btree(ntfs_volume *vol)
+{
+	ntfs_inode *ni;
+	struct dir *dir;
+	INDEX_ROOT *ir;
+	INDEX_ENTRY *next;
+	ntfs_attr_search_ctx *ctx = NULL;
+	ntfs_index_context *ictx = NULL;
+	ntfs_attr *bm_na = NULL;
+	int ret;
+	COLLATION_RULES cr;
+
+	dir = (struct dir *)calloc(1, sizeof(struct dir));
+	if (!dir) {
+		ntfs_log_error("Failed to allocate for subdir.\n");
+		return 1;
+	}
+
+	ni = ntfsck_check_root_inode(vol);
```

Comment:
root inode should be checked in ntfsck_check_system_files()...

Reply:
then it should check mft record bitmap for root inode in ntfsck_check_system_files().
I'll remove it.